src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1165,32 +1165,38 @@ object Desugar {
               _ <- write(in.New(target, zero)(src))
             } yield target
 
-          case PPredConstructor(base, args) => base match {
-            case PFPredBase(id) =>
-              val proxy = fpredicateProxyD(id)
-              val idT = info.typ(id) match {
+          case PPredConstructor(base, args) =>
+            def handleFPredBase(p: PPredConstructorBase): Writer[in.Expr] = {
+              val proxy = fpredicateProxyD(p.id)
```

Comment:
I am surprised that this works if the base is a predicate from another package.

Reply:
Do you think I should change it to something else?

Reply:
One thing I also tried was to implement `handleFPredBase` like the following but that triggers a `NodeNotInTreeException` for predicates from other packages

```scala
            def handleFPredBase(p: PPredConstructorBase): Writer[in.Expr] = p match {
              case PFPredBase(id) =>
                val proxy = fpredicateProxyD(id)
                val predConstructType = info.typ(id)
                val idT = predConstructType match {
                  case FunctionT(args, AssertionT) =>
                    in.PredT(args.map(typeD(_, Addressability.rValue)(src)), Addressability.rValue)
                  case t => violation(s"desugarer: $t cannot be converted to a predicate type")
                }
                for {
                  dArgs <- sequence(args.map { x => option(x.map(exprD(ctx)(_))) })
                } yield in.PredicateConstructor(proxy, idT, dArgs)(src)

              case PDottedBase(recvWithId) =>
                info.resolve(recvWithId) match {
                  case Some(ap.Predicate(id, symb)) =>
                    val proxy = fpredicateProxyD(symb.decl)
                    val predConstructType = info.typ(recvWithId)
                    // TODO: abstract this part
                    val idT = predConstructType match {
                      case FunctionT(args, AssertionT) =>
                        in.PredT(args.map(typeD(_, Addressability.rValue)(src)), Addressability.rValue)
                      case t => violation(s"desugarer: $t cannot be converted to a predicate type")
                    }
                    for {
                      dArgs <- sequence(args.map { x => option(x.map(exprD(ctx)(_))) })
                    } yield in.PredicateConstructor(proxy, idT, dArgs)(src)
                }
            }
```